common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java

```diff
@@ -132,10 +134,24 @@ public synchronized void saveAccountRecord(@NonNull AccountRecord accountRecord)
      * API to clear all cache.
      * Note: This method is intended to be only used for testing purposes.
      */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     public synchronized void clearCache() {
         getAccountCredentialCache().clearAll();
     }
 
+    /**
+     * API to inspect cache contents.
+     * Note: This method is intended to be only used for testing purposes.
+     *
+     * @return A immutable List of Credentials contained in this cache.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    public synchronized List<Credential> getCredentials() {
```

Comment:
Since we're adding this API, do we *need* to return a list?  Would it be sufficient just to return an Iterator<Credential>, since the contains method on List is going to have to search it anyway?

Reply:
This certainly _could_ return an Iterator, though the immediate benefit of this change isn't obvious to me since basically the only thing I want to do with this collection is call `contains()` on it in a test

Reply:
Point.  This is only being added for testing.

Reply:
I think the implicit/broader point though is "do we need all these `List` functions in a world of immutable state?" which is a good point to consider esp. given the potential perks an `Iterator` can provide (lazy loading, etc).

So I'll decline this particular change, but ACK the sentiment and will ponder it henceforth if we add additional cache APIs